File: 1_weirdAlgorithm.cpp
#include <bits/stdc++.h>
using namespace std;

void weirdAlgorithm(long long &n)
{
    while (n != 1)
    {
        cout << n << " ";
        if (n % 2)
        {
            n = 3 * n + 1;
        }
        else
        {
            n = n / 2;
        }
    }
    cout << 1;
}

int main()
{
    long long n;
    cin >> n;
    weirdAlgorithm(n);
}

File: 1_weirdAlgorithm.py
def weirdAlgorithm(n: int) -> None:
    while n != 1:
        print(n, end=" ")
        if n % 2:
            n = 3 * n + 1
        else:
            n = n // 2
    print(n, end=" ")


if __name__ == "__main__":
    n = int(input())
    weirdAlgorithm(n)
